packages/tpetra/core/src/Tpetra_CrsGraph_decl.hpp

```diff
@@ -590,9 +590,29 @@ namespace Tpetra {
       return copier_type::clone (*this, node2, params);
     }
 
+
     //! Destructor.
     virtual ~CrsGraph () = default;
 
+
+    /// \brief Swaps the data from *this with the data and maps from graph
+    ///
+    /// \param graph [in/out] a crsGraph
+    void swap(CrsGraph<LocalOrdinal, GlobalOrdinal, Node> & graph);
+
+
+    /// \brief True if and only if \c CrsGraph is exactly identical to this CrsGraph
+    ///
+    /// This performs _exact_ matches on objects with in the graphs. That is,
+    /// internal data structures such as arrays must match exactly in both
+    /// content and order. This is not performing any sort of isomorphic
+    /// search.
+    ///
+    /// \param graph [in] a crsGraph to compare against this one.
+    ///
+    bool isSameAs(const CrsGraph<LocalOrdinal, GlobalOrdinal, Node> &graph) const;
```

Comment:
Huh, I didn't realize you were going to make this a public method.  `isSameAs` should report graph equality, not data structure equality.  I would prefer that this method be made private until that can be fixed.

Reply:
I made this public because the other isSameAs that I saw (in MultiVector I think) was public.  

I've said this from the start that I'm not doing an isomorphism check, this is just checking that everything is the same in both graphs.

If I make this private or protected, how can I call it from the test?

Reply:
With changing the name to `isIdenticalTo`, I'm moving forward with keeping it public... let me know if we really do need to change that to protected or private.

@jhux2 I don't own that book... should I order a copy? Short term, do we have any examples showing this pattern in existing tests that I could mimic?

Reply:
@william76 I'd recommend getting a copy.  You might ask around in CSRI first to see if anyone has a copy you can browse.

Reply:
Make it public, call it something else, and add to the documentation that users should not call this function.